### PR TITLE
Addition of qvp functions from MeteoSwiss Py-ART + support of demixing in hydroclass_semisupervised

### PR DESCRIPTION
**Pull request by wolfidan**
Hi,

This PR aims at harmonizing a bit the MeteoSwiss fork with the official Py-ART. It adds several functions to qvp.py in retrieve submodules for quasi-vertical profiles, range-defined vertical profiles and enhanced vertical profiles (see docstrings for refs).

It also adds the demixing to the hydroclass_semisupervised function, allowing you to optionally get the entropy of the classification and the proportion of every hydrometeor class in the radar resolution volume. See https://amt.copernicus.org/articles/11/4847/2018/amt-11-4847-2018.html for more info.

- [x] Closes https://github.com/openradar/erad2024/issues/92
- [x] Tests added
- [x] Documentation reflects changes

### CODE REVIEW

**Comment by mgrover1** [comment]
@wolfidan thanks for submitting this so quickly! Can you please run pre-commit on the PR to resolve some of the linting issues?

```
pre-commit install

pre-commit run --all-files
```

**Comment by wolfidan** [comment]
Hi @mgrover1 , cool I learned the existence of thi pre-commit thing. With my black version the tests passed. Which version is used in the CI ?

File: pyart/core/transforms.py [hunk 1]
```diff
@@ -78,6 +78,35 @@ def antenna_to_cartesian(ranges, azimuths, elevations):
...
+
+    Returns
+    -------
+    ranges : array
+        Distances to the center of the radar gates (bins) in m.
+    azimuths : array
+        Azimuth angle of the radar in degrees. [-180., 180]
+    elevations : array
+        Elevation angle of the radar in degrees.
+    ke : float, optional
```
Comment (zssherman) [inline pyart/core/transforms.py:98]: I don't see ke in the returns list, did we want to remove it from the returns doc?

File: pyart/retrieve/qvp.py [hunk 2]
```diff
@@ -111,3 +141,1681 @@ def quasi_vertical_profile(
...
+            qvp.fields[field_name]["data"] = qvp_data.reshape(1, qvp.ngates)
+        else:
+            qvp.fields[field_name]["data"] = np.ma.concatenate(
+                (qvp.fields[field_name]["data"], qvp_data.reshape(1, qvp.ngates))
+            )
+
+    return qvp
+
+
+def compute_ts_along_coord(
```
Comment (zssherman) [inline pyart/retrieve/qvp.py:904]: Would it be possible to test the non qvp functions as well? I think we are ok on the private functions.

File: pyart/retrieve/qvp.py [hunk 3]
```diff
@@ -1,16 +1,47 @@
 ﻿"""
+pyart.retrieve.quasi_vertical_profile
```
Comment (zssherman) [inline pyart/retrieve/qvp.py:2]: I don't believe we need this module link anymore due to how the docs generate. We can still keep the "Retrieval of QVPs from a radar object.

**Comment by wolfidan** [comment]
@mgrover1 , sorry not sure how this works. I did what you told me but then when I do a git add, git commit, git push it just says Everything up-to-date.
Output of pre-commit run --all-files
was 

pre-commit run --all-files
trim trailing whitespace.................................................Passed
fix end of files.........................................................Passed
check docstring is first.................................................Passed
check json...............................................................Passed
check yaml...............................................................Passed
debug statements (python)................................................Passed
mixed line ending........................................................Passed
pyupgrade................................................................Failed
- hook id: pyupgrade
- exit code: 1
- files were modified by this hook

Rewriting pyart/retrieve/qvp.py

black....................................................................Failed
- hook id: black
- files were modified by this hook

reformatted pyart/retrieve/qvp.py

All done! ✨ 🍰 ✨
1 file reformatted, 240 files left unchanged.

File: tests/retrieve/test_echo_class.py [hunk 4]
```diff
@@ -144,26 +144,28 @@ def test_hydroclass_semisupervised():
...
 
     hydro = pyart.retrieve.hydroclass_semisupervised(
         radar_small, mass_centers=mass_centers
-    )
-    assert_allclose(hydro["data"][0][0:5], [6, 6, 6, 6, 6])
-    assert_allclose(hydro["data"][0][-5:], [2, 2, 2, 2, 2])
+    )["hydro"]
+
+    assert_allclose(hydro["data"][0][0:5], [7, 7, 7, 7, 7])
+    assert_allclose(hydro["data"][0][-5:], [3, 3, 3, 3, 3])
```
Comment (zssherman) [inline tests/retrieve/test_echo_class.py:169]: Are we able to test the two other types of dicts as well? Propx and entropy?

**Comment by zssherman** [comment]
I left a few comments, the example is failing as I think we need to plug in the hydro key in the example now that there is multiple fields present.

File: tests/retrieve/test_echo_class.py (hunk 4 above)
Comment (wolfidan) [reply tests/retrieve/test_echo_class.py:168]: Hi @zssherman ,
Thanks for the review!

I have added additional tests for all functions and for the additional outputs of hydroclass_semisupervised

**Comment by wolfidan** [comment]
I think the CI fails here because I try to get a key "hydroclass_entropy" from the pyart config file.  I have in fact committed a new default_config.py which adds the field hydroclass_entropy.
But I think the CI still uses the actual one (of the master branch) and so it crashes. So I have no idea how to fix this problem.
Do you have a clue?

**Comment by mgrover1** [comment]
It looks like tests are passing now 👍

**Comment by wolfidan** [comment]
There is one open point, we noticed a discrepancy in the standardization of RhoHV between our fork and the official repo. One of the boundary values is a bit different (it does not really impact on the classification but rather on the entropy and proportions). I am discussing with @jfigui to figure out the problem and which value is correct. I'll keep you posted very soon.

File: tests/retrieve/test_echo_class.py (hunk 4 above)
Comment (zssherman) [reply tests/retrieve/test_echo_class.py:196]: @wolfidan Awesome thank you!
